### PR TITLE
Rename dataset to dataset_iterator and other renames

### PR DESCRIPTION
**Pull request by seliverstov**

### CODE REVIEW

File: deeppavlov/dataset_readers/snips_intents_reader.py [hunk 1]
```diff
@@ -28,8 +28,8 @@
 log = get_logger(__name__)
 
 
-@register('classification_datasetreader')
-class ClassificationDatasetReader(DatasetReader):
+@register('snips_intents_reader')
+class SnipsIntentsDatasetReader(DatasetReader):
```
Comment (yoptar) [inline deeppavlov/dataset_readers/snips_intents_reader.py]: This class works with generic csv-files for classification, not just Snips dataset

File: deeppavlov/dataset_iterators/dstc2_dialog_iterator.py [hunk 2]
```diff
@@ -16,11 +16,11 @@
...
 
 from deeppavlov.core.common.registry import register
-from deeppavlov.core.data.dataset import Dataset
+from deeppavlov.core.data.dataset_iterator import BasicDatasetIterator
 
 
-@register('dialog_dataset')
-class DialogDataset(Dataset):
+@register('dstc2_dialog_iterator')
+class Dstc2DialogDatasetIterator(BasicDatasetIterator):
```
Comment (vikmary) [inline deeppavlov/dataset_iterators/dstc2_dialog_iterator.py]: The `DialogDataset`was intended to work not only with DSTC2 data, but any dialog data (babi, maluuba, etc.). So it preferably should be named `DialogDatasetIterator` or simular.

File: deeppavlov/dataset_iterators/dstc2_intents_iterator.py [hunk 3]
```diff
@@ -17,15 +17,15 @@
...
 from deeppavlov.core.common.log import get_logger
 
 
 log = get_logger(__name__)
 
 
-@register('intent_dataset')
-class IntentDataset(Dataset):
+@register('dstc2_intents_iterator')
+class Dstc2IntentsDatasetIterator(BasicDatasetIterator):
```
Comment (vikmary) [inline deeppavlov/dataset_iterators/dstc2_intents_iterator.py]: Same comment. Should be `IntentDatasetIterator`.

File: deeppavlov/dataset_iterators/dstc2_ner_iterator.py [hunk 4]
```diff
@@ -20,14 +20,14 @@
...
+from deeppavlov.core.data.dataset_iterator import BasicDatasetIterator
 from deeppavlov.core.data.utils import download
 
 logger = logging.getLogger(__name__)
 
 
-@register('dstc2_ner_dataset')
-class DstcNerDataset(Dataset):
+@register('dstc2_ner_iterator')
+class Dstc2NerDatasetIterator(BasicDatasetIterator):
```
Comment (vikmary) [inline deeppavlov/dataset_iterators/dstc2_ner_iterator.py]: The class is dependent on DSTC2 data, should remain `Dstc2NerDatasetIterator`.

File: deeppavlov/dataset_readers/snips_intents_reader.py (hunk 1 above)
Comment (seliverstov) [reply deeppavlov/dataset_readers/snips_intents_reader.py]: Ok, done.

File: deeppavlov/dataset_iterators/dstc2_ner_iterator.py (hunk 4 above)
Comment (seliverstov) [reply deeppavlov/dataset_iterators/dstc2_ner_iterator.py]: Ok

File: deeppavlov/dataset_iterators/dstc2_dialog_iterator.py (hunk 2 above)
Comment (seliverstov) [reply deeppavlov/dataset_iterators/dstc2_dialog_iterator.py]: Ok, done

**Pull request merged**